### PR TITLE
Run onModifyTemplate on rulesets for custom formats

### PR DESCRIPTION
**Pull request by MacChaeger**
And now to take my pound of flesh from battle performance for the sake of OM mashups 😈 

(but seriously this only runs once per switch and forme change)

### CODE REVIEW

**Comment by Zarel** [comment]
Meh, just use `runEvent`; it's fine.

**Comment by MacChaeger** [comment]
I already tried. It actually doesn't work because the rule-pseudoweathers aren't set when Pokemon get initialized.

**Comment by Zarel** [comment]
You could change that?

**Comment by MacChaeger** [comment]
oops

File: sim/battle.js [hunk 1]
```diff
@@ -77,6 +77,17 @@ class Battle extends Dex.ModdedDex {
...
+		for (const rule of this.getRuleTable(format).keys()) {
+			if (rule.startsWith('+') || rule.startsWith('-') || rule.startsWith('!')) continue;
+			let subFormat = this.getFormat(rule);
+			if (subFormat.exists) {
+				let hasEventHandler = Object.keys(subFormat).some(val =>
+					val.startsWith('on') && !['onBegin', 'onValidateTeam', 'onChangeSet', 'onValidateSet'].includes(val)
+				);
+				if (hasEventHandler) this.addPseudoWeather(rule);
+			}
+		}
```
Comment (Zarel) [inline sim/battle.js]: This whole block definitely goes at the end of `constructor`, right above the `setPlayer` calls.

**Comment by Zarel** [comment]
Okay, cool. I'm going to delay merging until after the TypeScript refactor is in, because that one's probably a harder merge conflict to deal with than this one.

**Comment by MacChaeger** [comment]
Looks like I botched the merge, hold on.

**Pull request merged**